### PR TITLE
docs(private-packages): add a note about Yarn 2+

### PR DESCRIPTION
**Pull request by ylemkimon**
## Changes:

Added a note about private packages in Yarn 2 to documentation.

## Context:

Renovate doesn't support reading `npmRegistries` and `npmScopes` from `.yarnrc.yml`, leading to confusion on how to configure Renovate on Yarn with private packages.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: docs/usage/getting-started/private-packages.md [hunk 1]
```diff
@@ -259,6 +259,12 @@ However be aware that if your `.npmrc` is too big to encrypt then the above comm
 }
 ```
 
+#### Yarn 2+
+
+Renovate doesn't support reading `npmRegistries` and `npmScopes` from `.yarnrc.yml`, so `hostRules` (or `npmToken`) and `npmrc` should be configured like above.
+Renovate updates `npmRegistries` in `.yarnrc.yml` with resolved `hostRules` before running Yarn.
+For Renovate to overwrite existing `npmRegistries` entry, the key should match the `matchHost` without the protocol (`http:` or `https:`) and with the trailing slash.
```
Comment (viceice) [inline docs/usage/getting-started/private-packages.md:266]: @rarkins Are we sure about this behavior?

Comment (rarkins) [reply docs/usage/getting-started/private-packages.md:266]: I didn't double check it but assumed @ylemkimon checked the source :)

Comment (viceice) [reply docs/usage/getting-started/private-packages.md:266]: I've verified, this won't work for npm datasource. The hostrule won't applied, as it's not a url or a hostname.

Comment (ylemkimon) [reply docs/usage/getting-started/private-packages.md:266]: I think we're talking about different things here. https://github.com/renovatebot/renovate/blob/62a540c272d4f0b2738781d0442d61fd946cb482/lib/util/url.ts#L49-L55 https://github.com/renovatebot/renovate/blob/62a540c272d4f0b2738781d0442d61fd946cb482/lib/manager/npm/post-update/rules.ts#L20-L21

I meant following Renovate configuration:
```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "<Encrypted PAT token>"
      }
    }
  ]
}
```
will update `.yarnrc.yml` as following:
```yaml
npmRegistries:
  //npm.pkg.github.com/:
    npmAuthToken: <Decrypted PAT token>
```

Comment (viceice) [reply docs/usage/getting-started/private-packages.md:266]: So `without` needs to be `with` `https?` prefix. I think the sample should be added too.

@ylemkimon can you create a new PR for it?

Comment (ylemkimon) [reply docs/usage/getting-started/private-packages.md:266]: I've opened #11696.

**Pull request merged**